tree.go

```diff
@@ -1110,6 +1116,10 @@ func (n *LeafNode) Commitment() *Point {
 }
 
 func (n *LeafNode) Commit() *Point {
+	if n.commitment == nil {
+		commitLeafNodes([]*LeafNode{n})
+	}
```

Comment:
Now `(*LeafNode).Commit()` can't assume that `n.commitment != nil`, we call `commitLeafNodes(...)` to do the proper calculation.